File: VB/View/TestView.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DragDropTwoGrids.ViewModels

Namespace DragDropTwoGrids.View

    Public Partial Class TestView
        Inherits UserControl

        Public Sub New()
            InitializeComponent()
            Dim helper As DragAndDropHelper = New DragAndDropHelper(gridControl1)
            If Not mvvmContext1.IsDesignMode Then InitBindings(helper)
        End Sub

        Private Sub InitBindings(ByVal helper As DragAndDropHelper)
            Dim fluentAPI = mvvmContext1.OfType(Of DragDropViewModel)()
            fluentAPI.SetBinding(gridControl1, Function(c) c.DataSource, Function(x) x.Files)
            fluentAPI.WithEvent(Of MyDragAndDropEventArgs)(helper, "Drop").EventToCommand(Sub(x) x.Drop(Nothing), Function(args) args.Record IsNot Nothing)
            fluentAPI.WithEvent(Of MyOnDeleteEventArgs)(helper, "RemoveRecord").EventToCommand(Sub(x) x.RemoveRecord(Nothing), Function(args) args.Record)
        End Sub
    End Class
End Namespace

File: CS/View/TestView.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DragDropTwoGrids.ViewModels;

namespace DragDropTwoGrids.View {
    public partial class TestView : UserControl {
        public TestView() {
            InitializeComponent();
            DragAndDropHelper helper = new DragAndDropHelper(this.gridControl1);
            if(!mvvmContext1.IsDesignMode)
                InitBindings(helper);
        }
        private void InitBindings(DragAndDropHelper helper) {
            var fluentAPI = mvvmContext1.OfType<DragDropViewModel>();
            fluentAPI.SetBinding(gridControl1, c => c.DataSource, x => x.Files);

            fluentAPI.WithEvent<MyDragAndDropEventArgs>(helper, "Drop")
                    .EventToCommand(x => x.Drop(null), args => args.Record != null);

            fluentAPI.WithEvent<MyOnDeleteEventArgs>(helper, "RemoveRecord")
                    .EventToCommand(x => x.RemoveRecord(null), args => args.Record);
        }
    }
}
